tests/linearizability/model.go

```diff
@@ -244,25 +289,76 @@ func applyRequestToSingleState(s EtcdState, request EtcdRequest) (EtcdState, Etc
 	s.KeyValues = newKVs
 	opResp := make([]EtcdOperationResult, len(request.Ops))
 	increaseRevision := false
+
 	for i, op := range request.Ops {
 		switch op.Type {
 		case Get:
 			opResp[i].Value = s.KeyValues[op.Key]
 		case Put:
 			s.KeyValues[op.Key] = op.Value
 			increaseRevision = true
+			s = detachFromOldLease(s, op)
 		case Delete:
 			if _, ok := s.KeyValues[op.Key]; ok {
 				delete(s.KeyValues, op.Key)
 				increaseRevision = true
+				s = detachFromOldLease(s, op)
 				opResp[i].Deleted = 1
 			}
+		case PutWithLease:
+			if _, ok := s.Leases[op.LeaseID]; ok {
```

Comment:
Question, does etcd allows to make a Put with lease that doesn't exist? I expect that it returns an client error, but maybe I remember wrong.

Reply:
Adding test result for quick ref -
Put with non existent lease -

{"level":"warn","ts":"2023-01-12T15:39:35.877-0800","caller":"txn/util.go:47","msg":"failed to apply request","took":"18.335µs","request":"header:<ID:7587867905305039876 > put:<key:\"x\" value_size:1 lease:4 >","response":"","error":"lease not found"}

client (etcdctl in my case) -
macbook % ./etcdctl put x y --lease=4
{"level":"warn","ts":"2023-01-12T15:39:35.878-0800","logger":"etcd-client","caller":"v3@v3.6.0-alpha.0/retry_interceptor.go:65","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc00017e000/127.0.0.1:2379","method":"/etcdserverpb.KV/Put","attempt":0,"error":"rpc error: code = NotFound desc = etcdserver: requested lease not found"}
Error: etcdserver: requested lease not found